src/components/sync-wizard/common/InstanceSelectionStep.tsx

```diff
@@ -13,19 +13,45 @@ export const getInstanceOptions = async (api: D2Api) => {
     }));
 };
 
+const getInstances = async (api: D2Api) => {
+    const { objects } = await Instance.list(api, {}, { paging: false });
+    return objects;
+};
+
 const InstanceSelectionStep: React.FC<SyncWizardStepProps> = ({ syncRule, onChange }) => {
     const d2 = useD2();
     const api = useD2Api();
     const [selectedOptions, setSelectedOptions] = useState<string[]>(syncRule.targetInstances);
     const [instanceOptions, setInstanceOptions] = useState<{ value: string; text: string }[]>([]);
+    const [targetInstances, setTargetInstances] = useState<Instance[]>([]);
+
+    const includeCurrentUrlAndTypeIsEvents = (selectedinstanceIds: string[]) => {
+        return (
+            syncRule.type === "events" &&
+            selectedinstanceIds
+                .map(id => targetInstances.find(instance => instance.id === id)?.url)
+                .includes(api.baseUrl)
```

Comment:
[minor] [untested]
```suggestion
            selectedinstanceIds
                .map(id => _.find(targetInstances, ["id", id])?.url)
                .includes(api.baseUrl)
```

Also this method signature has ``selectedinstanceIds: string[]``, isn't it the same as ``selectedOptions``? Is there a reason to use a block-scoped ``syncRule`` and a parameter ``selectedOptions``?

Reply:
Yes, It's the same but I need the value to invoke includeCurrentUrlAndTypeIsEvents function while selected instances are changing also if  I use selected options at this moment has not the last user action.

Regarding using lodash, I prefer not to have to use strings because it is more error-prone

Reply:
Actually with TS strings are type-checked and "id" in that example should be treated as a constant and the compiler normally verifies if the type includes a property called ``id``.

Reply:
There are many scenarios where to use strings is more error-prone, another example is rename refactor using rename symbol with vs code tool